components/match2/wikis/criticalops/match_group_input_custom.lua

```diff
@@ -0,0 +1,614 @@
+		player.name = player.name or Variables.varDefault(playerPrefix)
+		player.flag = player.flag or Variables.varDefault(playerPrefix .. 'flag')
+		player.displayname = player.displayname or Variables.varDefault(playerPrefix .. 'dn')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. count] = player
+			count = count + 1
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+
+-- Check if a map should be discarded due to being redundant
+-- DUMMY_MAP_NAME needs the match the default value in Template:Map
+function mapFunctions.discardMap(map)
+	return map.map == DUMMY_MAP_NAME
+end
+
+-- Parse extradata information
+function mapFunctions.getExtraData(map)
+	map.extradata = {
+		comment = map.comment,
+	}
+	return map
+end
+
+function mapFunctions._getHalfScores(map)
+	map.extradata.t1sides = {}
+	map.extradata.t2sides = {}
+	map.extradata.t1halfs = {}
+	map.extradata.t2halfs = {}
+
+	local key = ''
+	local overtimes = 0
+
+	local function getOppositeSide(side)
+		return side == SIDE_DEF and SIDE_ATK or SIDE_DEF
+	end
+
+	while true do
+		local t1Side = map[key .. 't1firstside']
+		if Logic.isEmpty(t1Side) or (t1Side ~= SIDE_DEF and t1Side ~= SIDE_ATK) then
+			break
+		end
+		local t2Side = getOppositeSide(t1Side)
+
+		-- Iterate over two Halfs (In regular time a half is 15 rounds, after that sides switch)
+		for _ = 1, 2, 1 do
+			if(map[key .. 't1' .. t1Side] and map[key .. 't2' .. t2Side]) then
+				table.insert(map.extradata.t1sides, t1Side)
+				table.insert(map.extradata.t2sides, t2Side)
+				table.insert(map.extradata.t1halfs, tonumber(map[key .. 't1' .. t1Side]) or 0)
+				table.insert(map.extradata.t2halfs, tonumber(map[key .. 't2' .. t2Side]) or 0)
+				map[key .. 't1' .. t1Side] = nil
+				map[key .. 't2' .. t2Side] = nil
+				-- second half (sides switch)
+				t1Side, t2Side = t2Side, t1Side
+			end
+		end
+
+		overtimes = overtimes + 1
+		key = 'o' .. overtimes
+	end
+
+	return map
+end
+
+-- Calculate Score and Winner of the map
+-- Use the half information if available
+function mapFunctions.getScoresAndWinner(map)
+	map.scores = {}
+	local indexedScores = {}
+
+	map = mapFunctions._getHalfScores(map)
+
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score
+		if Table.includes(ALLOWED_STATUSES, map['score' .. scoreIndex]) then
+			score = map['score' .. scoreIndex]
+		elseif Logic.isNotEmpty(map.extradata['t' .. scoreIndex .. 'halfs']) then
+			score = MathUtil.sum(map.extradata['t' .. scoreIndex .. 'halfs'])
+		else
+			score = tonumber(map['score' .. scoreIndex])
+		end
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			if TypeUtil.isNumeric(score) then
+				obj.status = STATUS_SCORE
+				obj.score = score
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = _NOT_PLAYED_SCORE
+			end
+			map.scores[scoreIndex] = score
+			indexedScores[scoreIndex] = obj
+		end
+	end
+
+	if map.finished == NP_MATCHSTATUS then
```

Comment:
comparing string with table

Reply:
Ok, this is the last unresolved stuff left for now

I currently had it this way (target is to get strikethrough back on **skip** maps)

```
if Table.includes(NOT_PLAYED_MATCH_STATUSES, map.finished) then
		map.resulttype = 'skip'
```

Should the map.resulttype became **Return table** or something to make strikethrough work?